Automattic/vip-go-mu-plugins
# Move out search settings health check/healing to dedicated cron job

## Description


## Changelog Description

### Search: Seperate content and settings validation/healing cron jobs

This change enables separate configuration for both types of health validations. While content validations run only on production environments and can be disabled by implementing a filter. Settings validation will run on all environments and can not be disabled.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Add a debug line so you can see output to: `search/includes/classes/class-settingshealthjob.php:84` - `var_dump($unhealthy_indexables);`
1. Run the cron action manually: `wp  eval "do_action('vip_search_settings_health');"`
1. No changes should be detected
1. Try changing something e.g.:  change `search/includes/classes/class-search.php:filter__ep_default_index_number_of_replicas`.
1. Run the cron action manually: `wp  eval "do_action('vip_search_settings_health');"`
1. The change should be reported
1. Run the cron action manually: `wp  eval "do_action('vip_search_settings_health');"` Again
1. No changes, as they were healt
